cmd/run/cmd.go

```diff
@@ -40,13 +40,15 @@ var Cmd = &cobra.Command{
 			os.Exit(1)
 		}
 
-		ctx, cancel := context.WithCancel(context.Background())
+		ctx, cancel := context.WithCancel(command.Context())
```

Comment:
The changes in this files are not needed for performance, but they made it slightly easier to test things, so I left them in.